drivers/gpio/gpio_bd8lb600fs.c

```diff
@@ -212,13 +230,33 @@ static int bd8lb600fs_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+	if (!gpio_is_ready_dt(&config->gpio_reset)) {
+		LOG_ERR("%s: reset GPIO is not ready", dev->name);
+		return -ENODEV;
+	}
+
 	int result = k_mutex_init(&drv_data->lock);
 
 	if (result != 0) {
 		LOG_ERR("unable to initialize mutex");
 		return result;
 	}
 
+	drv_data->instance_count_actual = config->gpios_count / 8;
+
+	if (config->gpios_count % 8 != 0) {
+		LOG_WRN("%s: number of GPIOs %i is not a multiple of 8, will round it up",
+			dev->name, config->gpios_count);
+		drv_data->instance_count_actual++;
+	}
+
+	if (drv_data->instance_count_actual > 4) {
+		LOG_WRN("%s: only a maximum of 4 devices can be daisy chained", dev->name);
+		drv_data->instance_count_actual = 32;
+	}
```

Comment:
Shouldn't this be an initialization error?

Reply:
Maybe a good idea to let it fail hard during initialization, I will change it.

Reply:
done